packages/core/src/use-cases/suppliers/update-supplier-use-case.ts

```diff
@@ -20,6 +20,14 @@ export class UpdateSupplierUseCase {
       throw new NotFoundError('Usuário não encontrado')
     }
 
+    const supplierEmail = supplierDto.email
+      ? await this.supplierRepository.findByEmail(supplierDto.email)
+      : null
+
+    if (supplierEmail && supplierEmail.id !== supplierId) {
```

Comment:
This is unnecessary as the email in the request will always be different from the one the supplier currently has.